packages/server/parseBody.ts

```diff
@@ -9,7 +9,7 @@ const parseBody = ({
   res: HttpResponse
   parser?: (buffer: Buffer) => any
 }) => {
-  return new Promise<JSON | null>((resolve) => {
+  return new Promise<JSON | Buffer | null>((resolve) => {
```

Comment:
this is a little brittle. the return type shall always be the return type of `parser` or `null`.

### Typescript Challenge for @tianrunhe @nickoferrall @tiffanyhan 

 try to rewrite this signature using a generic so whatever parser you pass into it, this function returns the return value of that generic or `null`.

for example:
```ts
const parser = (buffer: Buffer) => buffer.length === 0
const foo = await parseBody({res, parser}) // foo is boolean | null
```
 
The answer is here (no peeking until you're ready!): https://gist.github.com/mattkrick/c816bf3428b2ba47aa4d586bc55c8a4e